### PR TITLE
Bump UR tag to include PrintTrace fix

### LINKED ISSUE #14704: Plugin/level_zero_batch_barrier.cpp

### LINKED ISSUE #14721: Plugin/level_zero_dynamic_batch_test.cpp

### PR DESCRIPTION
**Pull request by callumfare**
Tests https://github.com/oneapi-src/unified-runtime/pull/1884

Fixes #14704 
Fixes #14721

### CODE REVIEW

**Comment by omarahmed1111** [comment]
@intel/dpcpp-esimd-reviewers Could we have a review on this when available ASAP to merge it? Thanks!

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake
Comment (omarahmed1111) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake]: Updated

**Review by sarnex — APPROVED** [review]

**Comment by omarahmed1111** [comment]
The bump for this PR will be included as part of this [PR](https://github.com/intel/llvm/pull/14701), so we could remove the UR bump here.

**Comment by callumfare** [comment]
> The bump for this PR will be included as part of this [PR](https://github.com/intel/llvm/pull/14701), so we could remove the UR bump here.

The test changes (other than two Plugin tests) are all needed at the same time as the bump, because the UR changes cause the test output to be formatted differently on L0. So that PR would need to temporarily disable those tests on some configurations, and then reenable them here. It might be easier to merge this separately.

**Comment by pbalcer** [comment]
> > The bump for this PR will be included as part of this [PR](https://github.com/intel/llvm/pull/14701), so we could remove the UR bump here.
> 
> The test changes (other than two Plugin tests) are all needed at the same time as the bump, because the UR changes cause the test output to be formatted differently on L0. So that PR would need to temporarily disable those tests on some configurations, and then reenable them here. It might be easier to merge this separately.

Makes sense. This PR is nearly finished with CI anyway. I think this should be merged first, followed by https://github.com/intel/llvm/pull/14701.


@intel/llvm-reviewers-runtime please review.

**Comment by pbalcer** [comment]
@intel/llvm-reviewers-runtime can you please approve? This PR only has test and UR related changes. The correct reviewer team should be @intel/unified-runtime-reviewers, but this PR was created before the `CODEOWNERS` file change.

File: sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp
Comment (maarquitos14) [inline sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp:2]: Not blocking the merge, but should this eventually be enabled? If so, please, create another issue for this.

**Review by maarquitos14 — APPROVED** [review]
LGTM

File: sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp
Comment (pbalcer) [reply sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp:2]: I think it was always like that, even before PI removal.

**Comment by pbalcer** [comment]
@intel/llvm-gatekeepers please merge

File: sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp
Comment (AlexeySachkov) [reply sycl/test-e2e/Plugin/level_zero_dynamic_batch_test.cpp:2]: @steffenlarsen and/or @lbushi25 are working on an ongoing effort to review all `XFAIL` and `UNSUPPORTED` directives in our e2e tests to add an explicit comment that those are either expected to be there due to our implementation status/device capabilities, or that those should be fixed as part of this or that tracker.

File: sycl/test-e2e/Basic/host-task-dependency.cpp
Comment (AlexeySachkov) [inline sycl/test-e2e/Basic/host-task-dependency.cpp:181]: This is not to block the PR, but my opinion is that we shouldn't have tests relying on tracing in `e2e` tests. Those should instead be refactored to be unit-tests. I've highlighted this already in PI removal PR and I would be happy to contribute to such rewrite myself (if only I had some time for it, sigh).

**Comment by omarahmed1111** [comment]
Included PRs in this UR bump:
* https://github.com/oneapi-src/unified-runtime/pull/1737
* https://github.com/oneapi-src/unified-runtime/pull/1790
* https://github.com/oneapi-src/unified-runtime/pull/1399
* https://github.com/oneapi-src/unified-runtime/pull/1908
* https://github.com/oneapi-src/unified-runtime/pull/1884

**Pull request merged**
